test/common/http/http2/http2_frame.cc

```diff
@@ -284,12 +285,12 @@ Http2Frame Http2Frame::makeMetadataFrameFromMetadataMap(uint32_t stream_index,
 
   auto payload_sequence = encoder.EncodeRepresentations(representations);
   ASSERT(payload_sequence->HasNext() || numberOfNameValuePairs == 0);
-  const size_t maxPayloadSize = 4 * 1024 * 1024;
-  std::string payload;
-  if (payload_sequence->HasNext()) {
-    payload = payload_sequence->Next(maxPayloadSize);
+  // Concatenate all the payload segments to a single string.
+  std::string payload("");
```

Comment:
I'm not sure the extra initialization parameter is useful. std::string is already initialized empty, right?

Reply:
Yeah, that's the same as the default c'tor initialization.
It's from an "old-school" view of c++ where it is better to have explicit initializations over leaving something that may be undefined.
Removed the explicit initialization to comply with newer style.